sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/Marshaller.cs

```diff
@@ -138,5 +137,13 @@ private static void SetUserAgentHeader(IRequestContext requestContext)
                 requestContext.Request.Headers[HeaderKeys.UserAgentHeader] = userAgent;
             }
         }
+
+        private static string ToUserAgentHeaderString(RequestRetryMode requestRetryMode)
+        {
+            if (requestRetryMode == RequestRetryMode.Adaptive)
```

Comment:
This change seems a bit limited since we have `Legacy` mode and we may add other retry modes

Reply:
I'm sorry, @muhammad-othman, but I don't understand your point.

`RequestRetryMode` has only 2 values: `Adaptive` and `Standard`. There's no `Legacy` value.

This code can be generated using something like [NetEscapades.EnumGenerators](https://andrewlock.net/netescapades-enumgenerators-a-source-generator-for-enum-performance/). This might me a bit more specific, but nothing that cannot be easily done. I just didn't want to taggle that in this PR.

Reply:
I've added a `Debug.Assert` to assert the value is the expected one, when in DEBUG.

Reply:
You need something that catches that at build-time, not at run-time.

Having `Debug.Assert` catches that at debug-time.

Comprehensive unit tests will also catch this kind of errors.

I don't mind pushing this to some utility class adding a comprehensive of unit tests, if you point me in the right direction.

Reply:
I was just thinking that `Debug.Assert` might be lost and someone adding a new retry mode will run atleast one request to test their changes, but I agree unit tests are much better.

You can use this utility method to make sure that the enum didn't change with a note to update `ToUserAgentHeaderString` if it changed.
https://github.com/aws/aws-sdk-net/blob/v4-development/sdk/test/Common/Utils/AssertExtensions.cs#L137
And here is an example of its usage 
https://github.com/aws/aws-sdk-net/blob/8e74d19cbc090d66def761fd5ae9bf01d76e6329/sdk/test/Services/SecurityToken/UnitTests/Custom/StaticCheckTests.cs#L44


And you can add the tests here
https://github.com/aws/aws-sdk-net/blob/v4-development/sdk/test/UnitTests/Custom/Runtime

Reply:
@muhammad-othman,

> And you can add the tests here https://github.com/aws/aws-sdk-net/blob/v4-development/sdk/test/UnitTests/Custom/Runtime

What solution would that be? There are 405 `.sln` files in the repo.

Reply:
Basically any service solution will include the core UnitTestUtilities porject, here is an example from S3

![image](https://github.com/user-attachments/assets/be45beb5-1c2c-4893-93a7-d62a54c60961)

Please include the newly add test file the following projects as we don't include all the files added to the Runtime folder by default
```
sdk\test\UnitTests\Custom\AWSSDK.UnitTestUtilities.NetFramework.csproj
sdk\test\UnitTests\Custom\AWSSDK.UnitTestUtilities.NetStandard.csproj
```

Reply:
After digging a little bit into the code, I realized that there is no validation when `IRequestContext.ClientConfig.RetryMode`, so I changed `ToUserAgentHeaderString` to have a fast path for know values and a fallback of `requestRetryMode.ToString().ToLowerInvariant()`:

```csharp
private static string ToUserAgentHeaderString(RequestRetryMode requestRetryMode)
{
    switch (requestRetryMode)
    {
        case RequestRetryMode.Standard:
            return "standard";
        case RequestRetryMode.Adaptive:
            return "adaptive";
        default:
            return requestRetryMode.ToString().ToLowerInvariant();
    }
}
```

Added this to sdk/test/Services/SecurityToken/UnitTests/Custom/StaticCheckTests.cs:

```csharp
[TestMethod]
public void LookForRequestRetryModeChanges()
{
    var expectedHash = "2FC699DB6A4284C5D53F3B7FA4E64165576FD1F6AE458BC737AE503F06DFBE4C";
    AssertExtensions.AssertEnumUnchanged(
        typeof(RequestRetryMode),
        expectedHash,
        "The Amazon.Runtime.Internal.Marshaller.ToUserAgentHeaderString method implementation may need to be updated.");
}
```